websites/T/TOD/presence.ts

```diff
@@ -0,0 +1,281 @@
+			setPresenceData() {
+				if (presence.isWatching()) {
+					const video = presence.getVideo();
+					if(presence.isLive()) {
+						presenceData.smallImageText = video.paused ? "Paused" : "Live";
+						presenceData.smallImageKey = video.paused
+							? Assets.Pause
+							: Assets.Live;
+
+					} else {
+						[, presenceData.endTimestamp] =
+						presence.getTimestampsfromMedia(video);
+
+						presenceData.smallImageText = video.paused ? "Paused" : "Playing";
+						presenceData.smallImageKey = video.paused
+							? Assets.Pause
+							: Assets.Play;
+					}
+
+					presenceData.details = presence.getTitle();
+					presenceData.state = "Match";
+
+					presenceData.buttons = [
+						{
+							label: "Watch Match",
+							url: document.URL,
+						},
+					];
+
+					if (video.paused) {
+						delete presenceData.startTimestamp;
+						delete presenceData.endTimestamp;
+					}
+				} else {
+					presenceData.details = "Watching match:";
+					presenceData.state = presence.getTitle();
+
+					presenceData.buttons = [
+						{
+							label: "Watch Match",
+							url: document.URL,
+						},
+					];
+				}
+			},
+		},
+		"/sports/event/soccer-([a-zA-Z0-9-]+)": {
+			setPresenceData() {
+				if (presence.isWatching()) {
+					const video = presence.getVideo();
+					[, presenceData.endTimestamp] =
+						presence.getTimestampsfromMedia(video);
+
+					presenceData.details = presence.getTitle();
+					presenceData.state = "Match";
+
+					presenceData.smallImageText = video.paused ? "Paused" : "Playing";
+					presenceData.smallImageKey = video.paused
+						? Assets.Pause
+						: Assets.Play;
+
+					presenceData.buttons = [
+						{
+							label: "Preview Match",
+							url: document.URL,
+						},
+					];
+
+					if (video.paused) {
+						delete presenceData.startTimestamp;
+						delete presenceData.endTimestamp;
+					}
+				} else {
+					presenceData.smallImageKey = Assets.Viewing;
+					presenceData.smallImageText = "Viewing";
+					presenceData.details = "Previewing match:";
+					presenceData.state = presence.getTitle();
+
+					presenceData.buttons = [
+						{
+							label: "Preview Match",
+							url: document.URL,
+						},
+					];
+				}
+			},
+		},
+		"/account": {
+			setPresenceData() {
+				presenceData.details = "Viewing their settings";
+			},
+		},
+	};
+
+	data.settings = [
+		{
+			id: "timestamp",
+			data: ["startTimestamp", "endTimestamp"],
+		},
+		{
+			id: "buttons",
+			data: ["buttons"],
+		},
+		{
+			id: "smallImage",
+			data: ["smallImageKey"],
+		},
+	];
+
+	let presenceSelect;
+
+	for (const [pathname, PData] of Object.entries(data.presence)) {
+		if (new RegExp(pathname).test(document.location.pathname)) {
+			presenceSelect = pathname;
+			PData.setPresenceData();
+		}
+	}
+
+if (!presenceSelect && presence.isWatching()) {
+    const videoType = presence.getVideoType();
+    const presenceKey = videoType === "movie" || videoType === "show" ? "/movie/([a-zA-Z0-9-]+)" : "Unknown";
```

Comment:
Since `videoType` is going to be `"show" | "movie"`, this should always be `"/movie/([a-zA-Z0-9-]+)"`